components/MultiPanel/index.ts

```diff
@@ -1,12 +1,197 @@
   }
 
   connectedCallback () {
+    this._childrenChange();
+  }
+  
+  // Click event handler
+  _onClick(event:Event) {
+    const el: Element = event.target;
+    this._expand(this._getClosestHeading(el));
+  }
+
+  // KeyDown event handler
+  _onKeyDown(event) {
+    const selectedEl : Element = event.target;
+    
+    // if keydown event is not on heading element, ignore
+    if (!this._getClosestHeading(selectedEl)) {
+      return;
+    }
+
+    // don’t handle modifier shortcuts used by assistive technology.    
+    if (event.altKey) {
+      return;
+    }
+    let newHeading;
+    switch (event.key) {
+      case 'ArrowLeft':
+      case 'ArrowUp':
+        newHeading = this._prevHeading();
+        break;
+
+      case 'ArrowRight':
+      case 'ArrowDown':
+        newHeading = this._nextHeading();
+        break;
+
+      case 'Home':
+        newHeading = this._firstHeading();
+        break;
+
+      case 'End':
+        newHeading = this._lastHeading();
+        break;
+
+      case 'Enter':
+      case ' ':
+      case 'Spacebar':
+        this._expand(this._getClosestHeading(document.activeElement));
+        break;
+
+      // Any other key press is ignored and passed back to the browser.
+      default:
+        return;
+    }
+
+    event.preventDefault();
+    selectedEl.setAttribute('tabindex', '-1');
+    if (newHeading) {
+      newHeading.setAttribute('tabindex', '0');
+      newHeading.focus();
+    }
+  }
+
+  _expand (heading : Element) {
+
+    if (heading === undefined) {
+      return;
+    }
+
+    const content = heading.nextElementSibling;
+    
+    // heading elment should always have nextElementSibling (checked on _childrenChange)
+    // but in case it is null, return.
+    if (content === null) {
+      return;
+    }
+
+    // toggle expanded and aria-expanded attoributes
+    if (content.hasAttribute('expanded')) {
+      content.removeAttribute('expanded')
+      content.setAttribute('aria-expanded', 'false');
+    } else {
+      content.setAttribute('expanded', '') 
+      content.setAttribute('aria-expanded', 'true');  
+    }
+  }
+  
+  // children of multi-panel should always in even nuber (heading/content pair)
+  // if children are odd numbers, add a div at the end to prevent potential error.
+  _childrenChange () {
+    if (this.children.length % 2 !== 0) {
+      console.error('detected odd number of elements inside multi-panel, please make sure you have heading/content pair')
+    }
+
+    const children : Element[] = Array.from(this.children);
+    let tabindexAlreadySet : boolean = false;
+    let randomId : string;
+
+    children.forEach((child, i) => {
+      if(i % 2 === 0){
+        // even index = heading element
+
+        // if panel-heading class is already assigned to the element (= new child was added later),
+        // skip the ID/tabindex setting activities & just flag tabindexAlreadySet
+        if (!child.classList.contains('panel-heading')){
```

Comment:
If there's an if statement with a short part & a long part, I think it's more readable to handle the short part first. In some cases it means you can avoid the `else`. In this case:

```js
if (child.classList.contains('panel-heading')) {
  tabIndexAlreadySet = true;
  return;
}

child.classList.add('panel-heading');
// …
```

The `return` will need to be `continue` if you change this to a for-loop.